_lib/wordpress_featured-topic_processor.py

```diff
@@ -0,0 +1,51 @@
+import sys
+import json
+import os.path
+import requests
+
+
+def posts_at_url(url):
+
+    current_page = 1
+    max_page = sys.maxint
+
+    while current_page <= max_page:
+
+        url = os.path.expandvars(url)
+        resp = requests.get(url, params={"page": current_page, "count": "-1"})
+        results = json.loads(resp.content)
+        current_page += 1
+        max_page = results["pages"]
+        total = 0
+        for p in results["posts"]:
+            total += 1
+            yield p
+
+
+def documents(name, url, **kwargs):
+
+    for post in posts_at_url(url):
+        yield process_post(post)
+
+
+def process_post(post):
+    post["_id"] = post["slug"]
+    post["author"] = [post["author"]["name"]]
+    names = ["og_title", "og_image", "og_desc", "twtr_text", "twtr_lang",
+             "twtr_rel", "twtr_hash", "utm_campaign", "utm_term",
+             "utm_content", "links"]
+    for name in names:
+        if name in post["custom_fields"]:
+            post[name] = post["custom_fields"][name]
+    if "links" not in post["custom_fields"]:
+        links = []
+        for x in xrange(0, 10):
```

Comment:
I know this is derived from old code, but I'm not sure why we are using `xrange` here (and in other processors).  The benefits for using `xrange` come when iterating over a huge range, and when we're expecting to `break` partway through, neither of which appear to be the case.